assets/common/src/log_utils/exceptions.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Exceptions util."""
+
+import time
+import logging
+from functools import wraps
+from azureml._common.exceptions import AzureMLException
+from azureml._common._error_definition.azureml_error import AzureMLError  # type: ignore
+from azureml._common._error_definition.system_error import ClientError  # type: ignore
+
+
+class ModelImportErrorStrings:
+    """Error strings."""
+
+    LOG_SAFE_GENERIC_ERROR = "{pii_safe_message:log_safe}"
+    LOG_UNSAFE_GENERIC_ERROR = "An error occurred: [{error}]"
+    VALIDATION_ERROR = "Error while validating parameters [{error:log_safe}]"
+    INVALID_HUGGING_FACE_MODEL_ID = (
+        "Invalid Hugging face model id: {model_id}."
+        " Please ensure that you are using a correct and existing model ID."
+    )
+    ERROR_FETCHING_HUGGING_FACE_MODEL_INFO = "Error in fetching model info for {model_id}. Error [{error}]"
+    BLOBSTORAGE_DOWNLOAD_ERROR = "Failed to download artifacts from {uri}. Error: [{error}]"
+    GIT_CLONE_ERROR = "Failed to clone {uri}. Error: [{error}]"
+
+
+class ModelImportException(AzureMLException):
+    """Base exception for Model Import handling."""
+
+    def __init__(self, exception_message, **kwargs):
+        """Initialize a new instance of LLMException.
+        :param exception_message: A message describing the error
+        :type exception_message: str
+        """
+        super(ModelImportException, self).__init__(exception_message, **kwargs)
+
+    @property
+    def error_code(self):
+        """Return error code for azureml_error."""
+        return self._azureml_error.error_definition.code
+
+
+class ModelImportError(ClientError):
+    """Internal Import Model Generic Error."""
+
+    @property
+    def message_format(self) -> str:
+        """Message format."""
+        return ModelImportErrorStrings.LOG_UNSAFE_GENERIC_ERROR
+
+
+class GITCloneError(ClientError):
```

Comment:
Lets add error classes specific to components we have in here

Reply:
Added and implemented in register.py